contexts/PaymentContext/src/Domain/ChecksumGenerator.php

```diff
@@ -30,14 +33,13 @@ public function __construct( array $checksumCharacters ) {
 	 * @return string The checksum as a single character present in the constructors array argument
 	 */
 	public function createChecksum( string $string ): string {
-		$checksum = array_sum(
-			array_map(
-				'ord', // Returns the ASCII value of the character
-				str_split( md5( $string ) )
-			)
-		);
+		$checksum = hexdec( substr( md5( $this->normalizeString( $string ) ), 0, 8 ) );
 
 		return $this->checksumCharacters[$checksum % count( $this->checksumCharacters )];
 	}
 
+	private function normalizeString( string $string ): string {
+		return strtoupper( preg_replace( $this->cleanupPattern, '', $string ) );
```

Comment:
Intuitively, I think that

    $generator->createChecksum( 'AAAA-AAAA-' )

should give the same checksum as

    $generator->createChecksum( 'aaaa-aaaa-' )


But it does not. Is that a feature?

Reply:
It's a bug. Thank you for catching it!

Reply:
#964

Reply:
Ugh, should have been more careful with merge. The regex stuff is also broken.

Reply:
Fixed in https://github.com/wmde/FundraisingFrontend/pull/966